src/plugins/mer/mergeneraloptionswidget.cpp

```diff
@@ -58,13 +66,111 @@ MerGeneralOptionsWidget::MerGeneralOptionsWidget(QWidget *parent)
+
+    updateGui();
 }
 
 MerGeneralOptionsWidget::~MerGeneralOptionsWidget()
 {
+    MerSettings::deviceModelsReset();
+
     delete m_ui;
 }
 
+void MerGeneralOptionsWidget::onDeviceModelEdited()
+{
+    MerSettings::populateWithDeviceModels(m_ui->deviceModelComboBox);
+}
+
+void MerGeneralOptionsWidget::deviceModelDconfChanged(const QString &value)
+{
+    const QString name = m_ui->deviceModelComboBox->currentData().toString();
+    const MerEmulatorDeviceModel model = MerSettings::deviceModels().value(name);
+    QTC_ASSERT(!model.isNull(), return);
+
+    if (model.dconf() != value) {
+        MerSettings::insertDeviceModel(MerEmulatorDeviceModel(name,
+                                                              model.displayResolution(),
+                                                              model.displaySize(),
+                                                              value,
+                                                              model.isSdkProvided()));
+    }
+}
+
+void MerGeneralOptionsWidget::deviceModelScreenChanged()
+{
+    const auto name = m_ui->deviceModelComboBox->currentData().toString();
+    const MerEmulatorDeviceModel model = MerSettings::deviceModels().value(name);
+    QTC_ASSERT(!model.isNull(), return);
+
+    const QSize resolution {m_ui->screenResolutionWidthSpinBox->value(),
+                m_ui->screenResolutionHeightSpinBox->value()};
+    const QSize screen {m_ui->screenWidthSpinBox->value(),
+                m_ui->screenHeightSpinBox->value()};
+    if (model.displayResolution() != resolution
+            || model.displaySize() != screen) {
+        MerSettings::insertDeviceModel(MerEmulatorDeviceModel(name,
+                                                              resolution,
+                                                              screen,
+                                                              model.dconf(),
+                                                              model.isSdkProvided()));
+    }
+}
+
+void MerGeneralOptionsWidget::updateGui()
+{
+    if (m_ui->deviceModelComboBox->count() == 0)
+        return;
+
+    const int index = qMax(0, m_ui->deviceModelComboBox->currentIndex());
+    const QString name = m_ui->deviceModelComboBox->itemData(index).toString();
+    m_ui->deviceModelComboBox->setCurrentIndex(index);
+
+    const MerEmulatorDeviceModel selectedModel = MerSettings::deviceModels().value(name);
+    QTC_ASSERT(!selectedModel.isNull(), return);
+
+    const QSize resolution = selectedModel.displayResolution();
+    const QSize screen = selectedModel.displaySize();
+
+    m_optionsWidget->setOptions(selectedModel.dconf());
+
+    m_ui->removeProfileButton->setDisabled(selectedModel.isSdkProvided());
+    m_ui->deviceModelComboBox->setToolTip(selectedModel.isSdkProvided()
+                                          ? tr("SDK provided emulator profile")
+                                          : tr("User defined emulator profile"));
```

Comment:
This would be implemented by the MerDeviceModelComboBox class as well

Reply:
Implemented in new MerDeviceModelComboBox class.